src/VecSim/algorithms/hnsw/hnswlib_c.cpp

```diff
@@ -12,59 +11,44 @@
 using namespace std;
 using namespace hnswlib;
 
-struct HNSWIndex {
-    HNSWIndex(VecSimType vectype, VecSimMetric metric, size_t dim, size_t max_elements,
-              size_t M = HNSW_DEFAULT_M, size_t ef_construction = HNSW_DEFAULT_EF_C,
-              size_t ef_runtime = HNSW_DEFAULT_EF_RT);
-
-    VecSimIndex base;
-    unique_ptr<SpaceInterface<float>> space;
-    HierarchicalNSW<float> hnsw;
-};
-
-#ifdef __cplusplus
-extern "C" {
-#endif
+/******************** Ctor / Dtor **************/
```

Comment:
There is no destructor implementation here, it means that the default distructor is what we need here?

Reply:
it is just a comment for this section, so in the future, when needed we can put it here.
we can use `pragma region` also but I'm not sure it is supported everywhere so I didn't use it.